pandas/core/frame.py

```diff
@@ -231,6 +231,21 @@ class DataFrame(NDFrame):
     >>> df3 = DataFrame(np.random.randn(10, 5),
     ...                 columns=['a', 'b', 'c', 'd', 'e'])
 
+    Leaving dtype=None in constructor will infer a wider type than necessary
+    >>> df_cols = {'year':np.int32, 'month':np.int8}
+    >>> df = pd.DataFrame(columns=df_cols.keys(), dtype=None, index=range(10), data=-1)
+    >>> df.dtypes
+    month	int64
+    year	int64
+
+    A list/dict/Series/array-like is not allowed. Also behaves differently to read_csv(dtype)
```

Comment:
not sure what you are trying to show here, this is just confusing (after line 241)